src/main/java/com/synopsys/integration/jenkins/scan/bridge/BridgeDownloadManager.java

```diff
@@ -1,100 +1,162 @@
-        File installationDirectory = new File(synopsysBridgeInstallationPath);
-
+    public String getBridgeVersionFromVersionFile(String versionFilePath) {
         try {
-            String versionsFileContent = Files.readString(installationDirectory.toPath().resolve(versionFile));
-            Matcher matcher = Pattern.compile("Synopsys Bridge Package: (\\d+\\.\\d+\\.\\d+)")
-                    .matcher(versionsFileContent);
-
-            if (matcher.find()) {
-                return matcher.group(1);
+            FilePath file = new FilePath(workspace.getChannel(), versionFilePath);
+            if (file.exists()) {
+                String versionsFileContent = file.readToString();
+                Matcher matcher = Pattern.compile("Synopsys Bridge Package: (\\d+\\.\\d+\\.\\d+)")
+                        .matcher(versionsFileContent);
+
+                if (matcher.find()) {
+                    return matcher.group(1);
+                }
             }
-        } catch (Exception e) {
-            e.printStackTrace();
+        } catch (IOException | InterruptedException e) {
+            listener.getLogger().println("An exception occurred while extracting bridge-version from the 'versions.txt': " + e.getMessage());
         }
-
         return null;
     }
 
-    public List<String> getAllAvailableBridgeVersionsFromArtifactory(String url) {
-        List<String> availableVersions = new ArrayList<>();
+    public String getLatestBridgeVersionFromArtifactory(String bridgeDownloadUrl) {
+        String extractedVersionNumber = extractVersionFromUrl(bridgeDownloadUrl);
+        if(extractedVersionNumber.equals(ApplicationConstants.NOT_AVAILABLE)) {
+            String directoryUrl = getDirectoryUrl(bridgeDownloadUrl);
+            if(versionFileAvailable(directoryUrl)) {
+                String versionFilePath = downloadVersionFile(directoryUrl);
+                String latestVersion = getBridgeVersionFromVersionFile(versionFilePath);
 
-        try {
-            Document doc = Jsoup.connect(url).get();
-            Elements elements = doc.getElementsByTag("a");
-
-            for (Element element : elements) {
-                String content = element.text().endsWith("/") ? element.text().substring(0, element.text().length() - 1) : element.text();
-                String version = content.matches("^[0-9]+\\.[0-9]+\\.[0-9]+") ? content : null;
-                if (version != null) {
-                    availableVersions.add(version);
-                }
+                Utility.removeFile(versionFilePath, workspace, listener);
+
+                return latestVersion;
             }
-        } catch (Exception e) {
-            e.printStackTrace();
+            else {
+                return ApplicationConstants.NOT_AVAILABLE;
+            }
+        }
+        else {
+            return extractedVersionNumber;
         }
+    }
 
-        return availableVersions;
+    public String downloadVersionFile(String directoryUrl) {
```

Comment:
Rename it to refer to artifactory.

Reply:
Renamed - https://github.com/synopsys-sig/synopsys-security-scan-plugin/blob/dev/src/main/java/com/synopsys/integration/jenkins/scan/bridge/BridgeDownloadManager.java#L100